tests/scripts/auto-grow-storage.sh

```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/env bash
+
+#############
+# FUNCTIONS #
+#############
+
+function calculateSize() {
+   local currentsize=$2
+   local unit=$1
+   rawsizeValue=0
+
+   if [[ "$currentsize" == *"Mi" ]]
+   then
+      rawSize=$(echo $currentsize | sed -e 's/Mi//g')
+      unitSize="Mi"
+      rawsizeValue=$rawSize
+   elif [[ "$currentsize" == *"Gi" ]]
+   then
+      rawSize=$(echo $currentsize | sed -e 's/Gi//g')
+      unitSize="Gi"
+      rawsizeValue=$(( rawSize * 1000 ))
+   elif [[ "$currentsize" == *"Ti" ]]
+   then
+      rawSize=$(echo $currentsize | sed -e 's/Ti//g')
+      unitSize="Ti"
+      rawsizeValue=$(( rawSize * 1000000 ))
+   else
+      echo "Unknown unit of $unit : ${currentsize}"
+      echo "Supported units are 'Mi','Gi','Ti'"
+      exit 1
+   fi
+}
+
+function compareSizes() {
+   local newsize=$1
+   local maxsize=$2
+   calculateSize newsize $newsize
+   local newsize=$rawsizeValue
+   calculateSize maxsize $maxsize
+   local maxsize=$rawsizeValue
+   if [ "${newsize}" -ge "${maxsize}" ]
+   then
+      return "1"
+   else
```

Comment:
just return 0 at the end if you don't go through the first condition that returns 1

Reply:
Was not using shellcheck before, updated the syntax accordingly. 
Thankx.